### PR TITLE
chore(ci): add support for x86-64 regions only

### LINKED ISSUE #1950: Maintenance: Adding new AWS Regions to the Lambda Layers Deployment
### Summary

AWS recently launched new regions: 

Asia Pacific (Melbourne)
ap-southeast-4
Asia Pacific (Hyderabad)
ap-south-2
Europe (Spain)
eu-south-2
Europe (Zurich)
eu-central-2
Middle East (UAE)
me-central-1

[AWS Global Infrastructure Map](https://aws.amazon.com/about-aws/global-infrastructure/)

According to this [AWS blog post](https://aws.amazon.com/blogs/aws/aws-lambda-functions-powered-by-aws-graviton2-processor-run-your-functions-on-arm-and-get-up-to-34-better-price-performance/), graviton2 (arm64) is not yet available in Spain and Zurich.

### Why is this needed?

Deploy Lambda Layers to these new regions.

### Which area does this relate to?

_No response_

### Solution

_No response_

### Acknowledgment

- [X] This request meets [Lambda Powertools Tenets](https://awslabs.github.io/aws-lambda-powertools-python/latest/#tenets)
- [X] Should this be considered in other Lambda Powertools languages? i.e. [Java](https://github.com/awslabs/aws-lambda-powertools-java/), [TypeScript](https://github.com/awslabs/aws-lambda-powertools-typescript/)

### PR DESCRIPTION
**Pull request by rubenfonseca**
**Issue number:** #1950

### Changes

> Please provide a summary of what's being changed

This PR changes our Layer and Canary stack to support regions that only support x86_64 Lambdas. We couldn't find an easy and reliable way to
detect dinamically if a given region supports ARM64 Lambdas, so instead we maintain the map manually and indicate which regions don't support it.

This makes heavy use of CDK/CloudFormation Parameters and Conditions, so the same CloudFormation template can be used across all regions.

### User experience

> Please share what the user experience looks like before and after this change

After this change, we will generate Lambda Layers for regions that only support x86_64 Lambdas. We will need to update the documentation when the 
first Layer version is generated, and validate that the script to update the layer version in the docs works with the single-architecture regions.

### Regions that will be added by this PR

- ap-south-2
- ap-southeast-4
- eu-central-2
- eu-south-2
- me-central-1

## Checklist

If your change doesn't seem to apply, please leave them unchecked.

* [x] [Meet tenets criteria](https://awslabs.github.io/aws-lambda-powertools-python/#tenets)
* [x] I have performed a self-review of this change
* [x] Changes have been tested
* [x] Changes are documented
* [ ] PR title follows [conventional commit semantics](https://github.com/awslabs/aws-lambda-powertools-python/blob/develop/.github/semantic.yml)

<details>
<summary>Is this a breaking change?</summary>

**RFC issue number**:

Checklist:

* [ ] Migration process documented
* [ ] Implement warnings (if it can live side by side)

</details>

## Acknowledgment

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.

### CODE REVIEW

**Comment by rubenfonseca** [comment]
@leandrodamascena unfortunately there's no easy to way to test this. So please check the code to see if it makes sense, and after we merge we can try a couple of workflows to see if any change is needed.

File: layer/layer/layer_stack.py [hunk 1]
```diff
@@ -1,39 +1,58 @@
...
         scope: Construct,
         construct_id: str,
+        layer_version_name: str,
         powertools_version: str,
-        ssm_paramter_layer_arn: str,
-        ssm_parameter_layer_arm64_arn: str,
+        architecture: Optional[Architecture] = None,
         **kwargs
-    ) -> None:
+    ):
```
Comment (leandrodamascena) [inline layer/layer/layer_stack.py]: Just out of curiosity. Any reason to remove the `None` here?

**Comment by leandrodamascena** [comment]
> @leandrodamascena unfortunately there's no easy to way to test this. So please check the code to see if it makes sense, and after we merge we can try a couple of workflows to see if any change is needed.

The code makes perfect sense to me, but like you said: there's no way to test it and just looking at the code I have no idea if we need to change anything else.
So, I follow you: let's merge this PR and check for possible changes. I'll approve.

**Review by leandrodamascena — APPROVED** [review]
Approved.

File: layer/layer/layer_stack.py (hunk 1 above)
Comment (rubenfonseca) [reply layer/layer/layer_stack.py]: Sorry no, I'll re-add it to make it more explicit.

**Pull request merged**